packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts

```diff
@@ -0,0 +1,224 @@
+     */
+  readonly insightRuleState?: InsightRuleStates;
+}
+
+/**
+ * A Contributor Insights Rule managed by the CDK
+ */
+export class InsightRule extends InsightRuleBase {
+
+  /**
+     * Reference an existing Contributor Insights Rule defined outside of the CDK code, by name
+     *
+     * @param scope The parent creating construct (usually 'this')
+     * @param id The construct's name
+     * @param insightRuleName The Contributor Insights Rule's name
+     */
+  public static fromInsightRuleName(scope: Construct, id: string,
+    insightRuleName: string): IInsightRule {
+    class Import extends InsightRuleBase {
+      public readonly insightRuleArn = core.Stack.of(scope).formatArn({
+        service: 'cloudformation',
+        resource: 'insight-rule',
+        resourceName: insightRuleName,
+      });
+      public readonly insightRuleRuleName = insightRuleName;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+     * Reference an existing Contributor Insights Rule defined outside of the CDK code, by arn
+     *
+     * @param scope The parent creating construct (usually 'this')
+     * @param id The construct's name
+     * @param InsightRuleArn The Contributor Insights Rule's arn
+     */
+  public static frominsightRuleArn(scope: Construct, id: string,
+    InsightRuleArn:string): IInsightRule {
+    class Import extends InsightRuleBase {
+      public readonly insightRuleArn = InsightRuleArn;
+      public readonly insightRuleRuleName =
+      core.Stack.of(scope).parseArn(InsightRuleArn, ':').resourceName!;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+     * Arn of this Contributor Insights Rule
+     *
+     * @attribute
+     */
+  public readonly insightRuleArn: string;
+
+  /**
+     * Name of this Contributor Insights Rule
+     *
+     * @attribute
+     */
+  public readonly insightRuleRuleName: string;
+
+  private readonly ruleBody: string;
+  private readonly ruleState?: InsightRuleStates;
+
+
+  constructor(scope: Construct, id: string, props: InsightRuleProps) {
+    super(scope, id, {
+      physicalName: props.insightRuleName,
+    });
+
+    /**
+         * Validate the inputs,
+         * We shouldnt need to validate rule state,
+         * I know ruleName has some restrictions
+         * Rulebody I suppose can go through a json/yaml validator
+         */
+    this.ruleState = props.insightRuleState;
+    this.ruleBody = props.insightRuleBody;
+
+    const insightRule = new CfnInsightRule(this, 'Resource', {
+      ruleName: props.insightRuleName,
+      ruleBody: this.ruleBody,
+      ruleState: this.ruleState || InsightRuleStates.ENABLED,
+    });
+
+    /**
+         * The arn and name are both attributes of the CFN resource
+         * @see https://docs.aws.amazon.com/cdk/api/latest/python/aws_cdk.aws_cloudwatch/CfnInsightRule.html
+         */
+    this.insightRuleArn = insightRule.attrArn;
+    this.insightRuleRuleName = insightRule.ref;
```

Comment:
Could you explain why this is insightRule.ref instead insightRule.attr_rule_name?

Reply:
I did not notice that insight rules had an attr rule name. All other constructs use the .ref; however, insights rule is unique to most of the other constructs, as they have this attribute. I'll use "attr_rule_name" for now, as it seems to be the equivalent and is more readable